neo/rawio/bci2000rawio.py

```diff
@@ -0,0 +1,346 @@
+        try:
+            self._global_time = time.mktime(time.strptime(param_defs['StorageTime']['value'], time_formats[0]))
+        except:
+            self._global_time = time.mktime(time.strptime(param_defs['StorageTime']['value'], time_formats[1]))
+
+        # Save variables to make it easier to load the binary data.
+        self._read_info = {
+            'header_len': file_info['HeaderLen'],
+            'n_chans': file_info['SourceCh'],
+            'sample_dtype': {'int16': np.int16, 'int32': np.int32, 'float32': np.float32}.get(file_info['DataFormat']),
+            'state_vec_len': file_info['StatevectorLen'],
+            'sampling_rate': param_defs['SamplingRate']['value']
+        }
+        # Calculate the dtype for a single timestamp of data. This contains the data + statevector
+        self._read_info['line_dtype'] = [('raw_vector', self._read_info['sample_dtype'], self._read_info['n_chans']),
+                                         ('state_vector', np.uint8, self._read_info['state_vec_len'])]
+        import os
+        self._read_info['n_samps'] = int((os.stat(self.filename).st_size - file_info['HeaderLen'])
+                                         / np.dtype(self._read_info['line_dtype']).itemsize)
+
+        # TODO: Do we just memmap it now, or elsewhere?
+        self._memmap = np.memmap(self.filename, dtype=self._read_info['line_dtype'],
+                                 offset=self._read_info['header_len'], mode='r')
+
+    
+    def _segment_t_start(self, block_index, seg_index):
+        return 0.
+
+    def _segment_t_stop(self, block_index, seg_index):
+        return self._read_info['n_samps'] / self._read_info['sampling_rate']
+
+    def _get_signal_size(self, block_index, seg_index, channel_indexes=None):
+        return self._read_info['n_samps']
+    
+    def _get_signal_t_start(self, block_index, seg_index, channel_indexes):
+        return 0.
+    
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, channel_indexes):
+        if i_start is None: i_start = 0
+        if i_stop is None: i_stop = self._read_info['n_samps']
+        assert (i_start >= 0 and i_start <= self._read_info['n_samps']), "i_start outside data range"
+        assert (i_stop >= 0 and i_stop <= self._read_info['n_samps']), "i_stop outside data range"
+        if channel_indexes is None:
+            channel_indexes = np.arange(self.header['signal_channels'].size)
+        return self._memmap['raw_vector'][i_start:i_stop, channel_indexes]
+    
+    def _spike_count(self,  block_index, seg_index, unit_index):
+        return 0
+    
+    def _get_spike_timestamps(self,  block_index, seg_index, unit_index, t_start, t_stop):
+        return []
+    
+    def _rescale_spike_timestamp(self, spike_timestamps, dtype):
+        return spike_timestamps
+
+    def _get_spike_raw_waveforms(self, block_index, seg_index, unit_index, t_start, t_stop):
+        return np.array((0, 0, 0))
```

Comment:
When _spike_count return 0 you can to simplify have
```python
def _get_spike_timestamps(...):
    return None
def _rescale_spike_timestamp(...):
    return None
def _get_spike_raw_waveforms(...):
    return None

```
because normally this never will be called.